### PR TITLE
Implements ResultBufferElements

### LINKED ISSUE #52: Implement ResultBufferElements
Probably the most user friendly way to handle result buffer size for reads is to create a function for `ResultBufferElements` that returns the map of name -> {offset_elements, data_elements} just like [c++](https://docs.tiledb.io/en/stable/c-api.html#c.tiledb_query_set_buffer). We just have to store these `uint64 pointers` internally (similar to how we store a reference to the data slice so it doesn't get GC'ed).

This is a follow on issue after #51

### PR DESCRIPTION
**Pull request by antalakas**
Implements ResultBufferElements
SetBufferVar additionally return offsetSize
Tests are added

Closes #52

### CODE REVIEW

File: query.go [hunk 1]
```diff
@@ -673,15 +680,68 @@ func (q *Query) SetBufferVar(attribute string, offset []uint64,
...
+// from a read query.
+// This is a map from the attribute name to a pair of values.
+// The first is number of elements (offsets) for var size attributes, and the
+// second is number of elements in the data buffer. For fixed sized attributes
+// (and coordinates), the first is always 0.
+func (q *Query) ResultBufferElements() (map[string][2]uint64, error) {
+	elements := make(map[string][2]uint64, 0)
+
+	// Will need the schema to infer data type size for attributes
+	schema, _ := q.array.Schema()
```
Comment (Shelnutt2) [inline query.go]: We should check for error and return if we error on getting the schema

File: query.go [hunk 2]
```diff
@@ -673,15 +680,68 @@ func (q *Query) SetBufferVar(attribute string, offset []uint64,
...
+
+	for attributeName, v := range q.resultBufferElements {
+		// Skip coords
+		if attributeName == "__coords" {
```
Comment (Shelnutt2) [inline query.go]: We should return the coordinates with this function too. You can get the coordinates type (to compute elements) by checking the domain type from the array schema. Coordinate will always been fixed length.

Comment (Shelnutt2) [reply query.go]: Also there is `TILEDB_COORDS` for the coordinate name string.

File: query.go (hunk 1 above)
Comment (antalakas) [reply query.go]: checked

File: query.go (hunk 2 above)
Comment (antalakas) [reply query.go]: Coordinates are now included in the map returned

**Pull request merged**